prestoadmin/util/base_config.py

```diff
@@ -29,6 +29,44 @@
 from prestoadmin.util.exception import ConfigurationError
 
 
+def validate_integer(candidate, min, max):
+    try:
+        i = int(candidate)
+    except ValueError:
+        raise Exception("%s is not an integer" % (candidate,))
+
+    if i < min or i > max:
+        raise Exception("%s is out of range [%s, %s]" % (i, min, max))
+
+    return i
+
+
+def prompt_choices(choices):
+    i = -1
+    prompt_string = ""
+    for choice in choices:
+        i += 1
+        prompt_string += "%s\t%s\n" % (i, choice)
+    prompt_string += "Please enter a number between 0 and %s" % (i,)
+
+    answer = prompt(
+            prompt_string,
+            validate=lambda candidate: validate_integer(candidate, 0, i))
+
+    return choices[answer]
+
+
+class SetEnvConfigItem(object):
```

Comment:
What is this for? Presumably the java 8 home but I don't follow how it's needed.